### ISSUE TITLE
Unexpected Internal Software Issue (EX_SOFTWARE)

### ISSUE THREAD

**Issue by laserray777** [dotnet/templating#5725]
Gentlemen,
When issueing this dotnet command-line command
For details on the exit code, refer to https://aka.ms/templating-exit-codes#70
PS C:\Users\Administrator.lasercropslocal\source\repos\laserray777> dotnet new mvc --no-https --output Unitrends_Cloud_OpenVPN_query --frameworknet7.0
Unable to cast object of type 'System.Boolean' to type 'System.String'.
   at System.CommandLine.Binding.ArgumentConverter.GetValueOrDefault[T](ArgumentConversionResult result)
   at Microsoft.TemplateEngine.Cli.Commands.TemplateOptionResult.FromParseResult(TemplateOption option, ParseResult parseResult)
   at Microsoft.TemplateEngine.Cli.Commands.TemplateResult.FromParseResult(TemplateCommand templateCommand, ParseResult parseResult)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.CollectTemplateMatchInfo(InstantiateCommandArgs args, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, TemplateGroup templateGroup)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.HandleNoTemplateFoundResult(InstantiateCommandArgs args, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, TemplateGroup templateGroup, IReporter reporter)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.HandleTemplateInstantationAsync(InstantiateCommandArgs args, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, TemplateGroup templateGroup, CancellationToken cancellationToken)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.ExecuteIntAsync(InstantiateCommandArgs instantiateArgs, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, InvocationContext context)
   at Microsoft.TemplateEngine.Cli.Commands.BaseCommand`1.InvokeAsync(InvocationContext context)
Sincerely,
Raymond Gregowicz
laserray777@lasercrops.com

**Comment by GangWang01** [comment]
Not sure what the dotnet version is. If the version is 7.0.100, it's most likely the same issue as https://github.com/dotnet/templating/issues/4494#issuecomment-1254715878 which was fixed in next release.

**Comment by vlada-shubina** [comment]
@laserray777 please confirm the version of `dotnet` you are using (`dotnet --info` output). Thank you.

**Comment by j-tee** [comment]
> dotnet --info

$ dotnet --info
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  3.1.425 [C:\Program Files\dotnet\sdk]
  6.0.306 [C:\Program Files\dotnet\sdk]
  7.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by JanKrivanek** [comment]
@laserray777 - you should be able to proceed when separating `--framework` option and its value `net7.0` with a space:

```
dotnet new mvc --no-https --output Unitrends_Cloud_OpenVPN_query --framework net7.0
```

That being said - it's still a valid bug report, as templating engine fails during attempt to properly inform about unknown option (`--frameworknet7.0`). This issue is addressed in upcomming 7.0.2 release.

Please let us know if this is acceptable - so that we can close the item. Or let us know if you have other concerns.

Thank you.

**Comment by YuliiaKovalova** [comment]
Hi @laserray777,

The invalid output message for your input was fixed for 7.0.2xx + a workaround is available (please check the response above), so I close the ticket.
Feel free to comment on it if you have any questions.

Thank you!
